cyclonedx/utils/conda.py

```diff
@@ -41,26 +41,23 @@ class CondaPackage(TypedDict):
     name: str
     platform: str
     version: str
-    md5_hash: str
+    md5_hash: Optional[str]
 
 
-def parse_conda_json_to_conda_package(conda_json_str: str) -> Union[CondaPackage, None]:
+def parse_conda_json_to_conda_package(conda_json_str: str) -> Optional[CondaPackage]:
     try:
         package_data = json.loads(conda_json_str)
-    except JSONDecodeError:
-        print(f'Failed to decode JSON: {conda_json_str}')
-        raise ValueError(f'Invalid JSON supplied - cannot be parsed: {conda_json_str}')
+    except JSONDecodeError as e:
+        raise ValueError(f'Invalid JSON supplied - cannot be parsed: {conda_json_str}') from e
 
-    if 'md5_hash' not in package_data.keys():
-        package_data['md5_hash'] = None
-
-    if isinstance(package_data, dict):
-        return CondaPackage(**package_data)
+    if not isinstance(package_data, dict):
+        return None
 
-    return None
+    package_data.setdefault('md5_hash', None)
+    return CondaPackage(package_data)  # type: ignore # @FIXME write proper type safe dict at this point
```

Comment:
I thought this was good enough
what do you think, @madpah ?

Reply:
Can't think of another way without more invasive changes for now. Happy enough if you are @jkowalleck!